src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/SemanticTokensEditsDiffer.cs

```diff
@@ -67,37 +67,59 @@ private Container<SemanticTokensEdit> ProcessEdits(IReadOnlyList<DiffEdit> diffs
                         }
                         else
                         {
-                            results.Add(new SemanticTokensEdit
+                            razorResults.Add(new RazorSemanticTokensEdit
                             {
                                 Start = diff.Position,
-                                Data = ImmutableArray<int>.Empty,
                                 DeleteCount = 1,
                             });
                         }
                         break;
                     case DiffEdit.Type.Insert:
                         if (current != null &&
-                            current.Data.HasValue &&
-                            current.Data.Value.Any() &&
+                            current.Data != null &&
+                            current.Data.Any() &&
                             current.Start == diff.Position)
                         {
-                            current.Data = current.Data.Append(NewArray[diff.NewTextPosition!.Value]).ToImmutableArray();
+                            current.Data.Add(NewArray[diff.NewTextPosition!.Value]);
```

Comment:
😮

Reply:
N arrays for a total size of N!, where N is the character length of the pasted text 🤦. What was I thinking?